platform/security/certificate/src/main/java/org/codice/ddf/certificate/CertificateManager.java

```diff
@@ -0,0 +1,267 @@
+     * <p/>
+     * Post-condition: The X509 returned is signed using the privateKey privKey
+     *
+     * @param signingRequest an initialized signing request
+     * @param privKey        the PrivateKey used to sign the certificate
+     * @return X509Certificate that has been signed with the supplied PrivateKey
+     * -------------------------------------------------------------------
+     */
+
+    public static X509Certificate signX509CertificateSHA256RSA(
+            X509v3CertificateBuilder signingRequest, PrivateKey privKey) {
+        X509Certificate certificate = null;
+        try {
+            byte[] certBytes = signingRequest.build(new SHA256RSASigner(privKey)).getEncoded();
+            CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+            certificate = (X509Certificate) certificateFactory
+                    .generateCertificate(new ByteArrayInputStream(certBytes));
+        } catch (IOException | CertificateException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+
+        return certificate;
+    }
+
+    /**
+     * ---------------------------------------------------------------------
+     * Method: registerBCSecurityProvider
+     * <p/>
+     * Purpose:  Registers BouncyCastle as a java.Security provider
+     * <p/>
+     * Pre-condition:  None
+     * <p/>
+     * Post-condition: Bouncy Castle is added to the list of java.Security Providers
+     * <p/>
+     * Parameters:     None
+     * <p/>
+     * Returns:        None
+     * -------------------------------------------------------------------
+     */
+
+    public static void registerBCSecurityProvider() {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    /**
+     * ---------------------------------------------------------------------
+     * Method: getHostName
+     * <p/>
+     * Purpose:  Used to retrieve the hostname of the machine
+     * <p/>
+     * Pre-condition:  None
+     * <p/>
+     * Post-condition: Hostname is returned
+     * <p/>
+     * Parameters:     None
+     *
+     * @return String representing the hostname
+     * -------------------------------------------------------------------
+     */
+
+    public static String getHostName() {
+        String hostName = null;
+        try {
+            hostName = InetAddress.getLocalHost().getHostName();
+        } catch (UnknownHostException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        return hostName;
+    }
+
+
+    /*+----------------------------------------------------------------------
+    ||
+    ||  Class: SHA256RSASigner
+    ||
+    ||        Purpose:  Implementation of ContentSigner that uses SHa256withRSA. Used to sign X509 certificates
+    ||
+    ||     Interfaces:  ContentSigner
+    ||
+    ++-----------------------------------------------------------------------*/
+
+    private static class SHA256RSASigner implements ContentSigner {
```

Comment:
Can we make this a public, top-level class and move the signX509CertificateSHA256RSA() method in it?